docs/rfc/block-registration.md

```diff
@@ -0,0 +1,414 @@
+
+### Styles Variations
+
+*   Type: `array`
+*   Optional
+*   Localized: No
+*   Property: `styleVariations`
+
+```json
+{ 
+	"styleVariations": [ 
+		{ "name": "default", "label": "Default", "isDefault": true }, 
+		{ "name": "other", "label": "Other" }
+	]
+}
+```
+
+Block styles can be used to provide alternative styles to block. It works by adding a class name to the block's wrapper. Using CSS, a theme developer can target the class name for the style variation if it is selected.
+
+Plugins and Themes can also register [custom block style](https://wordpress.org/gutenberg/handbook/designers-developers/developers/filters/block-filters/#block-style-variations) for existing blocks.
+
+### Transforms
+
+* Type: `string`
+* Optional
+* Localized: No
+* Property: `transforms`
+
+```json
+{ "transforms": "my-block-transforms.js" }
+```
+
+This property is a pointer to a JavaScript file containing the save function of the block transforms. The save function defines the way in which the different attributes should be combined into the final markup, which is then serialized by Gutenberg into `post_content`.
+
+See the [Transforms](https://wordpress.org/gutenberg/handbook/designers-developers/developers/block-api/block-registration/#transforms-optional) documentation for more details.
+
+### Deprecated versions
+
+* Type: `object[]`
+* Optional
+* Localized: No
+* Property: `deprecated`
+
+```json
+{ "deprecated": [ {
+	"attributes": {},
+	"save": "my-deprecated-save.js",
+	"supports": {}
+} ] }
+```
+
+This property contains the definition of the deprecated versions of the block type. It is used to ensure that old blocks with old markup are not considered invalid.
+
+See the [Deprecated Blocks](https://wordpress.org/gutenberg/handbook/designers-developers/developers/block-api/block-deprecation/) documentation for more details.
+
+### Supports
+
+* Type: `object`
+* Optional
+* Localized: No
+* Property: `supports`
+
+Optional block extended support features. 
+
+See the [block supports](https://wordpress.org/gutenberg/handbook/designers-developers/developers/block-api/block-registration/#supports-optional) documentation page for more details.
+
+### Stylesheets
+
+* Type: `object`
+* Optional
+* Localized: No
+* Property: `stylesheets`
+
+```json
+{
+	"stylesheets": {
+		"main": "my-block-style.css",
+		"editor": "my-block-editor-style.css",
+		"theme": "my-block-theme-style.css"
+	}
+}
+```
+
+This property is a pointer to CSS files containing the CSS used for the block in different contexts.
+
+## Internationalization
+
+Localized properties are automatically wrapped in `_x` function calls on the backend and the frontend of WordPress. These translations are added as an inline script to the `wp-block-library` script handle in WordPress core or to the plugin's script handle when it defines metadata defintion.
+
+WordPress string discovery automatically includes these strings to the plugin's or core's domain name.
```

Comment:
I might suggest "translates ... using" in place of "includes ... to" ? If the idea is in describing how it applies the domain argument for translation functions.

I'm not sure if it's just myself, but I find use of "core" phrasing (at least in documentation) to maybe not be clear. I'd considered whether it might just be better to call it "the default domain", but I find that could also lead to a different confusion on whether it's expected to be provided by the plugin author.

It occurs to me that for the "core domain", it would just not be specified at all?

Finally, when referencing "domain", I don't know if calling it as "domain name" might cause some confusion around the [canonical understanding of a domain name](https://en.wikipedia.org/wiki/Domain_name), even if it's not technically inaccurate to call it as such. "text domain" is the [full name](https://www.gnu.org/software/gawk/manual/html_node/Explaining-gettext.html) of the concept, and seems suitable as an in-place replacement. Maybe even making it specific to the value provided in the plugin manifest.

```suggestion
WordPress string discovery automatically translates these strings using the `textdomain` value specified in the plugin header.
```